components/banners.js

```diff
@@ -0,0 +1,86 @@
+import Alert from 'react-bootstrap/Alert'
+import styles from './banners.module.css'
+import { useEffect, useState } from 'react'
+import { useMe } from '../components/me'
+import { useMutation } from '@apollo/client'
+import { WELCOME_BANNER_MUTATION } from '../fragments/users'
+import { useToast } from '../components/toast'
+
+export default function NewVisitorBanner () {
+  const today = new Date()
+  const sameDay = (a, b) => {
+    return Math.abs(a.getTime() - b.getTime()) < 1000 * 24 * 60 * 60
+  }
+  const me = useMe()
+  const toaster = useToast()
+  const [state, setState] = useState(null)
+  const handleClose = async () => {
+    const obj = { date: state?.date || new Date().toJSON(), hidden: true }
+    const str = JSON.stringify(obj)
+    window.localStorage.setItem('newVisitorInfo', str)
+    setState(obj)
+    if (me) {
+      let error
+      try {
+        ({ error } = await hideWelcomeBanner({ variables: { } }))
+      } catch (e) {
+        error = e
+      }
+      if (error) toaster.danger(error.toString())
+    }
+  }
+  const [hideWelcomeBanner] = useMutation(WELCOME_BANNER_MUTATION, {
+    update (cache) {
+      cache.modify({
+        id: `User:${me.id}`,
+        fields: {
+          hideWelcomeBanner () {
+            return true
+          }
+        }
+      })
+    }
+  })
+  useEffect(() => {
+    let str = window.localStorage.getItem('newVisitorInfo')
+    let obj = JSON.parse(str)
+    if (!str) {
+      obj = { date: new Date().toJSON(), hidden: false }
+      str = JSON.stringify(obj)
+      window.localStorage.setItem('newVisitorInfo', str)
+    }
+    if (me && (me.hideWelcomeBanner || !sameDay(new Date(me.createdAt), today))) {
+      obj.hidden = true
+      str = JSON.stringify(obj)
+      window.localStorage.setItem('newVisitorInfo', str)
```

Comment:
These two lines are in both `if` clauses so you can just remove them in both and put them outside